plutus-contract/src/Plutus/Trace/Emulator/Extract.hs

```diff
@@ -78,7 +79,8 @@ writeScriptsTo ScriptsConfig{scPath, scCommand} prefix trace emulatorCfg = do
             case Aeson.eitherDecode bs of
                 Left err -> putStrLn err
                 Right pp ->
-                    let params = (_params emulatorCfg) { pProtocolParams = pp, pNetworkId = networkId }
+                    let params = (_params emulatorCfg) & protocolParamsL .~ pp
+                                                       & networkIdL .~ networkId
```

Comment:
It's not for just lenses in this case, at least in the `protocolParamsL` case: when we use `protocolParamsL` as a setter, we rebuild the corresponding `PParams`, so using the lens "hide" the call to `fromProtocolParams`.  I was considering the use of a smart constructor that uses `ProtocolParameters` instead of `PParams` and yes, it may be clearer.